core/Application/EnvironmentManipulator.php

```diff
@@ -0,0 +1,27 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Application;
+
+use Piwik\Application\Kernel\GlobalSettingsProvider;
+
+/**
+ * Used to manipulate Environment instances before the container is created.
+ * Only used by the testing environment setup code, shouldn't be used anywhere
+ * else.
+ */
+interface EnvironmentManipulator
+{
+    /**
+     * Create a custom GlobalSettingsProvider kernel object, overriding the default behavior.
+     *
+     * @param array $kernelObjects
+     * @return GlobalSettingsProvider
+     */
+    public function makeGlobalSettingsProvider(array $kernelObjects);
```

Comment:
I tried to anticipate possible future scenarios. Just in case, to make it easier for other devs.

No one should use EnvironmentManipulators, only code that should is TestingEnvironment.php.

I can modify the signature to `makeGlobalSettingsProvider()` to be as simple as possible.